tasks/setup.py

```diff
@@ -277,6 +266,353 @@ def setupPulumiConfig(config):
         warn(f"Expecting one of [true, false], got {logs_to_std_err}")
 
 
+def resolve_keypair_opts(
+    config: Config,
+    keypair_name: Optional[str] = None,
+    key_type: Optional[str] = None,
+    key_format: Optional[str] = None,
+    private_key_path: Optional[Path | str] = None,
+    public_key_path: Optional[Path | str] = None,
+    require_key_type: Optional[bool] = False,
+    require_keyfile_exists: Optional[bool] = False,
+):
+    """
+    Resolve key pair options, in decreasing order of priority:
+    - user input (parameters)
+    - local config
+    - defaults
+
+    Returns a dict with the resolved values.
+    """
+    if config.configParams is None or config.configParams.aws is None:
+        raise Exit("Config is missing aws section")
+    awsConf = config.configParams.aws
+
+    # defaults
+    if awsConf.keyPairName:
+        default_keypair_name = awsConf.keyPairName
+    else:
+        default_keypair_name = getpass.getuser()
+    if awsConf.privateKeyPath:
+        default_private_key_path = awsConf.privateKeyPath
+    else:
+        default_private_key_path = None
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I would be more opinionated here and use the value we suggest in the documentation, `id_rsa`

Reply:
this is done further down, and takes the key type, keypair name, and key format into account
https://github.com/DataDog/test-infra-definitions/pull/1045/files#diff-c17ad71d631afbbe74eea29844a37a1501410870263f1ea09d58f4a786e91772R326-R328

Reply:
Oh I see, so by default now the key is `username`.pem. This might conflict with other users keys, it changes from the current default of using `id_rsa.pem` for the path and `<username>` for the keypair name. Could we keep the current behaviour?